server/cluster/cluster_worker.go

```diff
@@ -223,3 +223,8 @@ func (c *RaftCluster) HandleBatchReportSplit(request *pdpb.ReportBatchSplitReque
 		zap.Int("total", last))
 	return &pdpb.ReportBatchSplitResponse{}, nil
 }
+
+// HandleBucketHeartbeat processes RegionInfo reports from client
+func (c *RaftCluster) HandleBucketHeartbeat(buckets *metapb.Buckets) error {
+	return c.processBucketHeartbeat(buckets)
```

Comment:
Why do we need this abstruction?

Reply:
In the first, HandleBucketHeartbeat should handle the change of the bucket meta(keys and version), but in the second , it will handle the statistics.